### ISSUE TITLE
issue: object converted to array when field is unregistered

### ISSUE THREAD

**Issue by Ploppy3** [react-hook-form/react-hook-form#8315]
### Version Number

7.30.0

### Codesandbox/Expo snack

https://codesandbox.io/s/crimson-monad-tpicxn?file=/src/App.js

### Steps to reproduce

1.Click on 'clear'
2.The `names` field is now an array

### Expected behaviour

It should still be an object

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Ploppy3** [comment]
I am trying to remove a field from on object of my form.
I don't think I'm doing something right though.

What I want is to remove a field that is controlled by a component while keeping the component on screen:

```js
{
  features: {
    3: { /*...*/ },
    5: { /*...*/ },
  },
}
```

```js
{
  features: {
    5: { /*...*/ },
  },
}

**Comment by bluebill1049** [comment]
> I think that I should set the value to null. Can you please confirm that it is the right way to do it?

That's correct.

also, the number index is used for the array, this is by design so we can do a type check against for your input name.

**Comment by Ploppy3** [comment]
> also, the number index is used for the array, this is by design so we can do a type check against your input name.

I'm not sure to understand, I thought the type would be determined based on `defaultValues`.

**Comment by bluebill1049** [comment]
> I'm not sure to understand, I thought the type would be determined based on defaultValues.

not really, the index number should be reserved for the array index. This is by design so we have a consistent way to type check field path.
